compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/intrinsics/SignatureString.kt

```diff
@@ -20,15 +22,9 @@ import org.jetbrains.kotlin.resolve.jvm.AsmTypes
 object SignatureString : IntrinsicMethod() {
     override fun invoke(expression: IrFunctionAccessExpression, codegen: ExpressionCodegen, data: BlockInfo): PromisedValue? {
         val function = (expression.getValueArgument(0) as IrFunctionReference).symbol.owner
-
-        // TODO do not use descriptors
-        val declaration = codegen.context.referenceFunction(
-            DescriptorUtils.unwrapFakeOverride(function.descriptor).original
-        ).owner
-
-        val method = codegen.context.methodSignatureMapper.mapAsmMethod(declaration)
+        val resolved = if (function is IrFunctionImpl) function.collectRealOverrides().first() else function
```

Comment:
Maybe use `is IrSimpleFunction` to avoid checking against particular implementation?

Reply:
Thanks! Done.